index.html

```diff
@@ -2532,183 +2532,169 @@ <h1>Servient Implementation</h1>
+                The details of such operations vary among different implementations.
+                A lifecycle management (LCM) system may encapsulate those lifecycle operations within a <a>Servient</a>
+                and use internal interfaces to realize the lifecycle management.
+                The <a>WoT Scripting API</a> includes one possible implementation of the LCM functionality.
+            </p>
+            <p>
+                Southbound, the <a>WoT Runtime</a> interfaces with the protocol stack implementation of the <a>Servient</a>
+                and decouples the behavior implementation from the details of the <a>Protocol Bindings</a>.
+                The <a>WoT Runtime</a> usually also interfaces with the underlying system,
+                for instance, to access local hardware or even proprietary communication protocols.
+                Both interfaces are implementation-specific, but must follow the <a>Thing</a> abstraction,
+                so that the behavior implementation need not depent on system-specific API calls.
+            </p>
+            <p>
+                Northbound, the <a>WoT Runtime</a> interfaces with the behavior implementation,
+                in this case through the <a>WoT Scripting API</a>.
+                <a href="#alt-servient-impl"></a> intruduces cases, where the optional building block is not used.
             </p>
         </section>
         <section>
-            <h2>Private Security Metadata</h2>
-            <p> Private security metadata is also conceptually
-                managed by the <a>WoT Runtime</a> but is intentionally not made
-                directly accessible to the application. In fact, in the
-                most secure hardware implementations, such private
-                security data is stored in a separate, isolated memory
-                and only an abstract set of operations (possibly even
-                implemented by an isolated processor and software stack)
-                is provided that limit the attack surface and prevent
-                external disclosure of this data. Private security data
-                is used transparently by the <a>Protocol Binding</a> to
-                authorize and protect the integrity and confidentiality
-                of interactions.</p>
+            <h2>Scripting Runtime</h2>
+            <p>
+                When the <a>WoT Scripting API</a> is used, the <a>Servient</a> requires a language runtime to actually manage and execute scripts.
+                This <a>Scripting Runtime</a> is part of the <a>WoT Runtime</a>, but is functionally distinct from it.
+                While the <a>WoT Runtime</a> implements the <a>Thing</a> abstraction, provides lifecycle management, and interfaces with the protocol stack and system,
+                the <a>Scripting Runtime</a> interprets the scripting language executes the application scripts (e.g., ECMAScript [[?ECMASCRIPT]]).
```

Comment:
See comment above on byte-code interpreters. The architecture should not preclude portable *applications*, that may be written in other languages.

Reply:
Where is the relation to portability here?

Machine code with dynamic linking would be just as "portable" assuming everyone has that execution environment. The portability mentioned in other places is mainly about the common API (cf. Web browser APIs) and a single execution environment (cf. Web apps in JavaScript).

Will fix the missing "and" and remove the ECMAScript reference.